### DISCUSSION TITLE
Erro Campo Double

### DISCUSSION THREAD

**Discussion by Russimar** [viniciussanchez/dataset-serialize#156] [Ideas]
Boa tarde Pessoal,

Estava com problema na seguinte situação: {"id": 1, "preco": ""} 
Campo preço no meu dataset está como Float, na unit DataSet.Serialize.Import, function JSONObjectToDataSet:
linha: 376 -  LTryStrToFloat := StrToFloat(LJSONValue.Value, LFormatSettings); Nessa situação acontece um erro na conversão.
Arrumei da seguinte forma:                     
LTryStrToFloat := StrToFloatDef(LJSONValue.Value,0, LFormatSettings);

Vinicuis, se tiver outra maneira, e puder me orientar te agradeço.

**Comment by viniciussanchez** [comment]
Acho que o problema ai principal seria se seu campo preço é um campo numérico, porque seu JSON está retornando ele como uma string?
Deveria no mínimo vir null ou então 0.
Mas interessante a sua solução, teria que ver se vamos ter outros problemas ao utilizar ela

**Reply by Russimar** [reply]
O meu servidor recebe o Json, mas o sistema que enviou não tratou ele como um numérico, enviou como se fosse string, mesmo que na documentação indique que é um campo valor. Resolvi tratar no servidor para não ter futuros problemas.
Valeuuu e Obrigado.
